sql/postgres/sqlspec.go

```diff
@@ -488,3 +452,33 @@ func attr(typ *schemaspec.Type, key string) (*schemaspec.Attr, bool) {
 	}
 	return nil, false
 }
+
+func typeSpec(t schema.Type) (*schemaspec.Type, error) {
+	if t, ok := t.(*schema.TimeType); ok && t.T != TypeDate {
+		spec := &schemaspec.Type{T: timeAlias(t.T)}
+		if t.Precision != defaultTimePrecision {
+			spec.Attrs = []*schemaspec.Attr{specutil.LitAttr("precision", strconv.Itoa(t.Precision))}
+		}
+		return spec, nil
+	}
+	s, err := FormatType(t)
+	if err != nil {
+		return nil, err
+	}
+	return &schemaspec.Type{T: s}, nil
+}
+
+// formatTime overrides the default printing logic done by schemahcl.hclType.
+func formatTime() specutil.TypeSpecOption {
```

Comment:
Should the driver know about hcl?

Reply:
This is why I added the `Printer` prop you removed in this PR @a8m

Reply:
The driver knows nothing about hcl. It's the same schemaspec API that exists today. However, since there is some reflection/parsing logic going on there, I added this comment to make the life easier for future readers. 

I don't think the Printer is needed. From its name, I assumed the printer was used for printing, but that's was not the case (print to driver presentation and then call Parse). Instead, I'd add something like `WithTypeParser` to allow overriding the parsing logic of one typespec if we'll need something like this to avoid one layer of abstraction.